src/main/webapp/assets/js/feedScript.js

```diff
@@ -921,3 +994,94 @@ function toggleLegend() {
     legend.style.display = 'none';
   }
 }
+
+/**
+ * Shows/Hides the filters to the user.
+ */
+function toggleFilters() {
+  const filterCard = document.getElementById('filter-card');
+
+  if (getComputedStyle(filterCard, null).display === 'none') {
+    filterCard.style.display = 'block';
+    toggleFiltersButton.style.backgroundColor = 'white';
+    toggleFiltersButton.style.color = 'var(--dark-maroon)';
+    toggleFiltersButton.style.border = '1px solid var(--dark-maroon)';
+  } else {
+    filterCard.style.display = 'none';
+    // Using '' makes the colors fall back to CSS defaults
+    toggleFiltersButton.style.backgroundColor = '';
+    toggleFiltersButton.style.color = '';
+    toggleFiltersButton.style.border = '';
+  }
+}
+
+/**
+ * Filters the posts the user sees on the page based
+ * on the current state of the filter inputs.
+ */
+function filterAndUpdatePagePosts() {
+  // Grab the filter inputs
+  const filters = {
+    numPeople: parseInt(document.getElementById('num-people').value),
+    happeningNow: document.getElementById('happening-now').checked,
+    distance: parseFloat(document.getElementById('distance').value),
+    keywords: document.getElementById('keywords').value === '' ?
+      [] : document.getElementById('keywords').value.split(',').map((e) => e.trim()),
+  };
+
+  const location = userLocation ?
+    {lat: userLocation.coords.latitude, long: userLocation.coords.longitude} :
+    null;
+
+  // Perform the filtering on the source of the page's posts
+  const filteredPosts = filterPosts(posts, new Date(), location, filters);
+
+  // Update the posts shown on the page
+  removePosts();
+  removeMarkers();
+  addPosts(filteredPosts);
+  addMarkers(filteredPosts);
+}
+
+/**
+ * Applies the filtering algorithm to an array of posts.
```

Comment:
Explain a bit more about this filtering algorithm. Things to talk about:

- How did you develop it?
- Are there any caveats/costs?
- Anything else an engineer should be wary of when reading this.